src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1011,17 +1011,19 @@ private function getValidatorMappingFilesFromDir($dir, array &$files)
         }
     }
 
-    private function getValidatorMappingFilesFromConfig(array $config, array &$files)
+    private function getValidatorMappingFilesFromConfig(ContainerBuilder $container, array $config, array &$files)
     {
         foreach ($config['mapping']['paths'] as $path) {
-            if (is_dir($path)) {
-                $this->getValidatorMappingFilesFromDir($path, $files);
-            } elseif (is_file($path)) {
-                if (preg_match('/\.(xml|ya?ml)$/', $path, $matches)) {
-                    $extension = $matches[1];
-                    $files['yaml' === $extension ? 'yml' : $extension][] = $path;
-                } else {
-                    throw new \RuntimeException(sprintf('Unsupported mapping type in "%s", supported types are XML & Yaml.', $path));
+            if ($container->fileExists($path)) {
```

Comment:
content needs to be tracked for directories (but only for new XML and YAML files). But it does not need to be tracked for files (changing the content of the validation file does not need to invalidate the container). So this shoould be improved.